homeassistant/components/tesla_fleet/climate.py

```diff
@@ -254,13 +260,7 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
         if not (temp := kwargs.get(ATTR_TEMPERATURE)):
             return
 
-        if temp == 30:
-            cop_mode = CabinOverheatProtectionTemp.LOW
-        elif temp == 35:
-            cop_mode = CabinOverheatProtectionTemp.MEDIUM
-        elif temp == 40:
-            cop_mode = CabinOverheatProtectionTemp.HIGH
-        else:
+        if (cop_mode := TEMP_LEVELS.get(temp)) is None:
```

Comment:
Are users expected to know the 3 specific temperatures here? I'm curious how this is expected to work. Should set temp be supported at all or might more preset modes make sense here?

Reply:
The Home Assistant UI knows the minimum and maximum and step, so no the user doesn't need to know because that information is communicated to them through the climate card UI.

I feel it's better to communicate temperatures than Low/Medium/High because a user will likely not know what those translate to.

Reply:
I think i missed that `target_temperature_step` makes it work. I was expecting a ui where the user moves it up by 1 degree and it fails.